target/linux/ramips/image/mt7621.mk

```diff
@@ -915,6 +982,25 @@ define Device/d-team_pbr-m1
 endef
 TARGET_DEVICES += d-team_pbr-m1
 
+define Device/dna_valokuitu-plus-ex400
```

Comment:
As I understand it, DNA is a an ISP. I am wondering, whether we should instead opt for the vendor of the device instead (i.e. Inteno EX400) and add any other brandings such as "DNA" "Valokuitu Plus EX400" and later rename like "Genexis" "EX400" via DEVICE_ALT[0-4]_VENDOR and MODEL. I think for DTS we should then also go for the HW vendor for model and compatible. Let me know what you think.

https://genexis.eu/product/pulse-ex400/downloads/

Full disclosure: I work at IOPSYS (software part and sister company of what is Genexis (formerly Inteno)).  My opinion is my own, but I think it seems to in line with some other ISP devices that are used by multiple ISPs. Feel free to correct me if I am wrong.

Obviously, you cannot test with hardware that you do not have, but I can tell you the Hw-wise they are identical and I will also give it a test run on the generic variants without ISP branding this week. Googling for EX400 will yield you quite a few Operators who use EX400 without any white label branding on the case and thus use Inteno EX400 (or Genexis EX400 for some later revisions) in their instructions.

Reply:
Also, in case you are wondering about anything, feel free to ask

Reply:
> Yes, DNA is an ISP they have stuck their name on the device. I would not
> think it differs from the Genexis EX400 in anything else but the LED
> panel on the top. First off I started with name Genexis but renamed for
> DNA because that's what it says on the top of the device and I could be
> sure whether there are some other differences to them than just LED panel.

> In Finland you can find Inteno/Genexis EX400 in shops and DNA EX400
> solely in DNA webshop and they sell it with their broadband subscription

Oh, I was not aware of this variant. I was slightly wrong. We definitely need a different DTS for the two variants then, because the LEDs are different.  Let's leave things as they are then for now.

> Anyhow, I have no objection to making a dtsi, say inteno_ex400.dtsi and
> then just redefine manufacturer/model in a dna_ex400.dts. And then maybe
> a drop a WPS button, I assume Inteno model has one.

> One known issue I have is that I don't know where the WIFI MACs are
> stored and the same goes for their caldata or what it is they need for
> the radios to run. At the moment both radios just operate as usual
> without dts pointing to eeprom or such. Previously the firmware had a
> 'factory' partition but it got lost when iopsys/inteno shifted to UFI.

Give me some time to look into the different HW-variants and the Factory-partition question (and any other questions),  but I think that that any data (including MAC addresses) is stored in the OTP of the wifi chips themselves.

Reply:
Only difference between the two variants is the LED panel at the top.
There is no danger from running it on non-DNA except LED-behaviour. I will create an MR for the non-DNA branded variant once this one is merged.